packages/backend-plugin-api/src/services/system/types.ts

```diff
@@ -78,9 +83,14 @@ export interface InternalServiceFactory<
 export type ServiceFactoryOrFunction = ServiceFactory | (() => ServiceFactory);
 
 /** @public */
-export interface ServiceRefOptions<TService, TScope extends 'root' | 'plugin'> {
+export interface ServiceRefOptions<
+  TService,
+  TScope extends 'root' | 'plugin',
+  TInstances extends 'single' | 'multiple',
+> {
   id: string;
   scope?: TScope;
+  multiple?: TInstances extends 'multiple' ? true : false;
```

Comment:
Naming options:

- `collectAllImplementations: true`
- `useAllImplementations: true`
- `allowMultipleImplementations: true`
- `installations: 'multiple'`
- `installations: 'many'`
- `implementations: 'multiple'`
- `implementations: 'many'`
- `collected: true`
- `collect: true`
- `singleton: false`
- `instances: 'multiple'` // this one is a bit problematic because we talk about multiple instances for plugins scoped services, see [here](https://backstage.io/docs/backend-system/architecture/services#root-context-for-service-factories)